src/openMVG/sfm/pipelines/sequential/sequential_SfM.cpp

```diff
@@ -401,17 +405,10 @@ bool SequentialSfMReconstructionEngine::Process()
       char values[20];
       std::ifstream ifs;
 
-      if(!std::system("nproc >values.txt"))
-      {
-        ifs.open("values.txt");
-        if(ifs.is_open())
-        {
-          ifs.get(values, 20, '\n');
-          ifs.close();
-          _tree.put("hardware.cpu.cores", values);
-        }
-      }
+      // CPU cores
+      _tree.put("hardware.cpu.cores", sysconf(_SC_NPROCESSORS_ONLN));
 
+      // CPU frequency
```

Comment:
You can use cpufreq.h like that:
https://github.com/vmolsa/cpufreq-example/blob/master/cpu.c